### PR TITLE
Feature/2469 add schema version browsing

### PR DESCRIPTION
**Pull request by HrisyToteva**
closes dfds/cloudplatform/issues/2469

# Additional Review Notes

https://github.com/dfds/selfservice-portal/assets/58768740/539770fd-4f01-477b-b2d8-21f408bb85c1

### CODE REVIEW

File: src/src/pages/capabilities/KafkaCluster/MessageContract.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const MessageStatus = {
   REQUESTED: "Requested",
 };
 
-function MessageHeader({
+export function MessageHeader({
   messageType,
   isOpen,
   status,
```
Comment (kralle333) [inline src/src/pages/capabilities/KafkaCluster/MessageContract.js:21]: Maybe we want to move this to some other file or even just MessageContracts and remove the export

File: src/src/pages/capabilities/KafkaCluster/MessageContracts.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  const headerClickHandler = () => {
+    if (onHeaderClicked) {
+      onHeaderClicked(messageType);
+      setSelectedContract(contracts.find((x) => x.Id === id));
+    }
+  };
+
+  useEffect(() => {
+    console.log(selectedContract);
+  }, [selectedContract]);
```
Comment (kralle333) [inline src/src/pages/capabilities/KafkaCluster/MessageContracts.js:70]: guess we can remove this.

File: src/src/pages/capabilities/KafkaCluster/MessageContracts.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        }}
+      >
+        {json}
+      </SyntaxHighlighter>
+    </div>
+  );
+}
+
+export default function MessageContracts({
+  id,
```
Comment (kralle333) [inline src/src/pages/capabilities/KafkaCluster/MessageContracts.js:33]: I dont see this set anywhere. Is it leftover from before?

File: src/src/pages/capabilities/KafkaCluster/MessageContracts.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      } else if (contracts.length > 1) {
+        setCanExpand(true);
+      }
+    }
+  }, [contracts]);
+
+  const headerClickHandler = () => {
+    if (onHeaderClicked) {
+      onHeaderClicked(messageType);
+      setSelectedContract(contracts.find((x) => x.Id === id));
```
Comment (kralle333) [inline src/src/pages/capabilities/KafkaCluster/MessageContracts.js:64]: id is not set, maybe we want to just save the ID of the first version.

File: src/src/pages/capabilities/KafkaCluster/MessageContract.module.css [hunk 5]
```diff
@@ -52,6 +52,17 @@
   background-color: #fff;
   border: 1px solid #4d4e4c29;
 
+  height: 43rem;
+  animation-name: slidein;
+  animation-duration: 0.5s;
+  animation-timing-function: ease;
+}
+
+.contentcontainer2 {
```
Comment (kralle333) [inline src/src/pages/capabilities/KafkaCluster/MessageContract.module.css:61]: can you give this a better name?

**Review by kralle333 — CHANGES_REQUESTED** [review]
Great job! I know you are in the process of removing the old message contract code, but once that has been done and the comments have been addressed lets get it in.

File: src/src/pages/capabilities/KafkaCluster/MessageContract.module.css (hunk 5 above)
Comment (HrisyToteva) [reply src/src/pages/capabilities/KafkaCluster/MessageContract.module.css:61]: Removed

File: src/src/pages/capabilities/KafkaCluster/MessageContracts.js (hunk 3 above)
Comment (HrisyToteva) [reply src/src/pages/capabilities/KafkaCluster/MessageContracts.js:33]: I am giving this to the MessageContracts as a param

File: src/src/pages/capabilities/KafkaCluster/MessageContracts.js (hunk 4 above)
Comment (HrisyToteva) [reply src/src/pages/capabilities/KafkaCluster/MessageContracts.js:64]: are we 100% sure that the first version will be always saved as SchemaVersion = 1

**Review by kralle333 — APPROVED** [review]
Looks good :) just show minimum version instead of schema version 1

**Pull request merged**